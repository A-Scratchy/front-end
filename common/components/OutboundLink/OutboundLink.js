import React from 'react';
import PropTypes from 'prop-types';
import ReactGA from 'react-ga';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import faExternalLinkAlt from '@fortawesome/fontawesome-free-solid/faExternalLinkAlt';
import styles from './OutboundLink.css';

OutboundLink.propTypes = {
  analyticsEventLabel: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  hasIcon: PropTypes.bool,
  href: PropTypes.string.isRequired,
};

OutboundLink.defaultProps = {
  className: '',
  hasIcon: true,
};

function OutboundLink({
  analyticsEventLabel, children, className, hasIcon, href,
}) {
  const linkContent = (
    <React.Fragment>
      <span className={styles.screenReaderOnly}>Opens in new window</span>
      {children}
      {hasIcon && (
        <FontAwesomeIcon
          className={styles.externalLinkIcon}
          icon={faExternalLinkAlt}
          style={{
            width: '14px',
          }}
        />
      )}
    </React.Fragment>
  );

  const analyticsMessage = `OUTBOUND [${analyticsEventLabel}] from ${window.location.pathname}`;

  if (process.env.NODE_ENV === 'production') {
    return (
      <ReactGA.OutboundLink
        className={className}
        eventLabel={analyticsMessage}
        rel="noopener noreferrer"
        target="_blank"
        to={href}
      >
        {linkContent}
      </ReactGA.OutboundLink>
    );
  }

  return (
    <a
      className={className}
      href={href}
      onClick={() => {
        // eslint-disable-next-line no-console
        console.log(`Analytics disabled. Message: ${analyticsMessage}`);
      }}
      rel="noopener noreferrer"
      target="_blank"
    >
      {linkContent}
    </a>
  );
}

export default OutboundLink;
