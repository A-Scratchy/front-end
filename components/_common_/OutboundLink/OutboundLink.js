import React from 'react';
import PropTypes from 'prop-types';
import ReactGA from 'react-ga';
import ExternalLinkIcon from 'static/images/icons/FontAwesome/external-link-square-alt-solid.svg';
import ScreenReaderOnly from 'components/_common_/ScreenReaderOnly/ScreenReaderOnly';
import styles from './OutboundLink.css';

OutboundLink.propTypes = {
  analyticsEventLabel: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  hasIcon: PropTypes.bool,
  href: PropTypes.string.isRequired,
};

OutboundLink.defaultProps = {
  className: undefined,
  hasIcon: true,
};

function OutboundLink({ analyticsEventLabel, children, className, hasIcon, href }) {
  return (
    <ReactGA.OutboundLink
      className={className}
      eventLabel={`OUTBOUND [${analyticsEventLabel}]`}
      rel="noopener noreferrer"
      target="_blank"
      to={href}
    >
      <>
        <ScreenReaderOnly>Opens in new window</ScreenReaderOnly>
        {children}
        {hasIcon && <ExternalLinkIcon className={styles.externalLinkIcon} />}
      </>
    </ReactGA.OutboundLink>
  );
}

export default OutboundLink;
