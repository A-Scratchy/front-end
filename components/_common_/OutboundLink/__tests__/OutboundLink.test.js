/* eslint-env jest */
import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';

import OutboundLink from '../OutboundLink';

describe('OutboundLink', () => {
  it('should render with required props', () => {
    createSnapshotTest(
      <OutboundLink analyticsEventLabel="Test" href="https://tests.com" router={MockedRouter}>
        Test
      </OutboundLink>,
    );
  });

  it('should render with many props assigned', () => {
    createSnapshotTest(
      <OutboundLink
        analyticsEventLabel="Test"
        className="test-class"
        href="https://tests.com"
        hasIcon={false}
        router={MockedRouter}
      >
        Test
      </OutboundLink>,
    );
  });
});
