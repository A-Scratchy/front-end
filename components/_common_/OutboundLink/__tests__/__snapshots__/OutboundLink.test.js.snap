// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutboundLink should render with all props assigned 1`] = `
<a
  className="test-class"
  href="https://tests.com"
  rel="noopener noreferrer"
  target="_blank"
>
  <span
    className="screenReaderOnly"
  >
    Opens in new window
  </span>
  Test
</a>
`;

exports[`OutboundLink should render with required props 1`] = `
<a
  className=""
  href="https://tests.com"
  rel="noopener noreferrer"
  target="_blank"
>
  <span
    className="screenReaderOnly"
  >
    Opens in new window
  </span>
  Test
  <test-file-mock
    className="externalLinkIcon"
  />
</a>
`;
